dom-fetch-client/src/main/scala/org/http4s/dom/FetchClientBuilder.scala

```diff
@@ -0,0 +1,146 @@
+    val credentials: Option[RequestCredentials],
+    val mode: Option[RequestMode],
+    val redirect: Option[RequestRedirect],
+    val referrer: Option[FetchReferrer],
+    val referrerPolicy: Option[ReferrerPolicy]
+)(implicit protected val F: Async[F])
+    extends BackendBuilder[F, Client[F]] {
+
+  private def copy(
+      requestTimeout: Duration = requestTimeout,
+      cache: Option[RequestCache] = cache,
+      credentials: Option[RequestCredentials] = credentials,
+      mode: Option[RequestMode] = mode,
+      redirect: Option[RequestRedirect] = redirect,
+      referrer: Option[FetchReferrer] = referrer,
+      referrerPolicy: Option[ReferrerPolicy] = referrerPolicy
+  ): FetchClientBuilder[F] =
+    new FetchClientBuilder[F](
+      requestTimeout,
+      cache,
+      credentials,
+      mode,
+      redirect,
+      referrer,
+      referrerPolicy
+    ) {}
+
+  def withRequestTimeout(requestTimeout: Duration): FetchClientBuilder[F] =
+    copy(requestTimeout = requestTimeout)
+
+  def withCacheOption(cache: Option[RequestCache]): FetchClientBuilder[F] =
+    copy(cache = cache)
+  def withCache(cache: RequestCache): FetchClientBuilder[F] =
+    withCacheOption(Some(cache))
+  def withDefaultCache: FetchClientBuilder[F] =
+    withCacheOption(None)
+
+  def withCredentialsOption(credentials: Option[RequestCredentials]): FetchClientBuilder[F] =
+    copy(credentials = credentials)
+  def withCredentials(credentials: RequestCredentials): FetchClientBuilder[F] =
+    withCredentialsOption(Some(credentials))
+  def withDefaultCredentials: FetchClientBuilder[F] =
+    withCredentialsOption(None)
+
+  def withModeOption(mode: Option[RequestMode]): FetchClientBuilder[F] =
+    copy(mode = mode)
+  def withMode(mode: RequestMode): FetchClientBuilder[F] =
+    withModeOption(Some(mode))
+  def withDefaultMode: FetchClientBuilder[F] =
+    withModeOption(None)
+
+  def withRedirectOption(redirect: Option[RequestRedirect]): FetchClientBuilder[F] =
+    copy(redirect = redirect)
+  def withRedirect(redirect: RequestRedirect): FetchClientBuilder[F] =
+    withRedirectOption(Some(redirect))
+  def withDefaultRedirect: FetchClientBuilder[F] =
+    withRedirectOption(None)
+
+  def withReferrerOption(referrer: Option[FetchReferrer]): FetchClientBuilder[F] =
+    copy(referrer = referrer)
+  def withReferrer(referrer: FetchReferrer): FetchClientBuilder[F] =
+    withReferrerOption(Some(referrer))
+  def withDefaultReferrer: FetchClientBuilder[F] =
+    withReferrerOption(None)
+
+  def withReferrerPolicyOption(referrerPolicy: Option[ReferrerPolicy]): FetchClientBuilder[F] =
+    copy(referrerPolicy = referrerPolicy)
+  def withReferrerPolicy(referrerPolicy: ReferrerPolicy): FetchClientBuilder[F] =
+    withReferrerPolicyOption(Some(referrerPolicy))
+  def withDefaultReferrerPolicy: FetchClientBuilder[F] =
+    withReferrerPolicyOption(None)
+
+  /** Creates a [[Client]].
+    */
+  def create: Client[F] = FetchClient.makeClient(
```

Comment:
Name it `build`? Like `EmberClientBuilder`.

Reply:
I followed the pattern from `JavaNetClientBuilder` and `OkHttpBuilder`, which have a `resource` method for building a `Resource` (the method is inherited from `BackendBuilder`), and `create` to return a `Client`.

I see that `EmberClientBuilder` does not inherit from `BackendBuilder` and `build` returns a `Resource`. Maybe we could make it a `BackendBuilder` subclass and rename `build` -> `resource` there? Let me know if you want me to add that to this PR.

Reply:
Yeah, there are some discrepancies between the various implementations of both the client and server builders. There is some work to try to clean that up, but for now it is still kind of a free for all.

Reply:
Aha, I see. I tend to look to ember for what to do since my understanding is that it's the [blessed](https://github.com/http4s/http4s/discussions/4140#discussioncomment-261676) server/client implementation. Sounds like this hopefully will be cleaned up and made more uniform as 1.0 finalizes.